index.js

```diff
@@ -42,21 +80,57 @@ const queryDns = (version, options) => {
 
 	const socketQuery = promisify(socket.query.bind(socket));
 
-	// eslint-disable-next-line promise/prefer-await-to-then
-	const promise = socketQuery({questions: [data.dnsQuestion]}, 53, data.dnsServer).then(({answers}) => {
-		socket.destroy();
+	const promise = (async () => {
+		let internalPromise = Promise.resolve();
+
+		data.dnsServers.forEach(dnsServerInfo => {
+			const {servers, question} = dnsServerInfo;
+			servers.forEach(server => {
+				// eslint-disable-next-line promise/prefer-await-to-then
```

Comment:
The rule is there for a reason. There's never a case where you have to use `.then`.

Reply:
I tried to do a for-in loop with await but linter was not happy with it ... in this case, I chain then after then so that they run concurrently to simulate a for loop with await inside ... maybe there's a better approach I'm not seeing